src/scipp/curve_fit.py

```diff
@@ -0,0 +1,515 @@
+    - The returned optimal parameter values popt and the coverance matrix pcov will
+      have units provided that the initial parameters have units. popt and pcov are
+      DataGroup and a DataGroup of DataGroup respectively. They are indexed using
+      the fit parameter names. The variance of the returned optimal parameter values
+      is set to the corresponding diagonal value of the covariance matrix.
+
+    Parameters
+    ----------
+    coords:
+        The coords that act as predictor variables in the fit.
+        If a mapping the keys signify names of arguments to f and the values
+        signify coordinate names in `da.coords`. If the fit coordinate
+        is not present in `da.coords` it can be passed as a Variable.
+    f:
+        The model function, f(x, y..., a, b...). It must take all coordinates listed
+        in `coords` as arguments, all other arguments will be treated as parameters
+        of the fit.
+    da:
+        The values of the data array provide the dependent data. If the data array
+        stores variances then the standard deviations (square root of the variances)
+        are taken into account when fitting.
+    p0:
+        An optional dict of optional initial guesses for the parameters. If None,
+        then the initial values will all be 1 (if the parameter names for the function
+        can be determined using introspection, otherwise a ValueError is raised). If
+        the fit function cannot handle initial values of 1, in particular for parameters
+        that are not dimensionless, then typically a :py:class:`scipp.UnitError` is
+        raised, but details will depend on the function.
+    bounds:
+        Lower and upper bounds on parameters.
+        Defaults to no bounds.
+        Bounds are given as a dict of 2-tuples of (lower, upper) for each parameter
+        where lower and upper are either both Variables or plain numbers.
+        Parameters omitted from the `bounds` dict are unbounded.
+    reduce_dims:
+        Additional dimensions to aggregate while fitting.
+        If a dimension is not in `reduce_dims` or in any of the dimensions
+        of the coords then that dimension will be present in the output
+        and one fit will be performed for every slice of that dimension.
+        If a dimension is passed to `reduce_dims` it will not be present in the output.
+    unsafe_numpy_f:
+        By default the provided fit function `f` is assumed to take scipp Variables as
+        input and use scipp operations to produce a scipp Variable as output.
+        This has the safety advantage of unit checking.
+        However, in some cases it might be advantageous to implement `f` using Numpy
+        operations. If `unsafe_numpy_f` is set to `True` then the arguments passed
+        to `f` will be Numpy arrays instead of scipp Variables.
+
+    Returns
+    -------
+    popt:
+        Optimal values for the parameters.
+    pcov:
+        The estimated covariance of popt.
+
```

Comment:
One final thing: Can you add a `See Also` section here and in `scipp.scipy.optimize.curve_fit` and reference each other?

Reply:
Should I refer to `scipp.scipy.optimize.curve_fit` as a function we keep for backwards compatibility?

Reply:
I'd just put a link with no comment?

Reply:
I think it is good to explain the difference between them in a short comment in the See Also. Then the user will have an easier time navigating the documentation.